File: scripts/renderGreekTexture.swift
import AppKit

let args = CommandLine.arguments
guard args.count >= 4 else {
    fputs("Usage: renderGreekTexture.swift <glyph> <name> <outPath>\n", stderr)
    exit(1)
}

let glyph = args[1]
let name = args[2]
let outPath = args[3]

let size: CGFloat = 256
let bgColor = NSColor(calibratedRed: 11/255, green: 17/255, blue: 34/255, alpha: 1)
let textColor = NSColor(calibratedRed: 250/255, green: 204/255, blue: 21/255, alpha: 1)

let image = NSImage(size: NSSize(width: size, height: size))
image.lockFocus()

bgColor.setFill()
NSBezierPath(rect: NSRect(x: 0, y: 0, width: size, height: size)).fill()

let fontSize: CGFloat = size * 0.65
let font = NSFont(name: "HelveticaNeue-Bold", size: fontSize) ?? NSFont.systemFont(ofSize: fontSize, weight: .bold)
let paragraph = NSMutableParagraphStyle()
paragraph.alignment = .center

let attributes: [NSAttributedString.Key: Any] = [
    .font: font,
    .foregroundColor: textColor,
    .paragraphStyle: paragraph
]

let attributed = NSAttributedString(string: glyph, attributes: attributes)
let textSize = attributed.size()
let rect = NSRect(x: 0, y: (size - textSize.height) / 2 - size * 0.05, width: size, height: textSize.height)
attributed.draw(in: rect)

image.unlockFocus()

guard let tiff = image.tiffRepresentation,
      let rep = NSBitmapImageRep(data: tiff),
      let png = rep.representation(using: .png, properties: [:]) else {
    fputs("Failed to render texture for \(name)\n", stderr)
    exit(2)
}

do {
    try png.write(to: URL(fileURLWithPath: outPath))
    print("Created \(outPath)")
} catch {
    fputs("Failed to write \(outPath): \(error)\n", stderr)
    exit(3)
}

File: scripts/render_greek_texture.py
#!/usr/bin/env python3
from AppKit import (
    NSImage,
    NSSize,
    NSColor,
    NSBezierPath,
    NSRect,
    NSFont,
    NSMutableParagraphStyle,
    NSAttributedString,
    NSBitmapImageRep,
    NSPNGFileType,
    NSFontAttributeName,
    NSForegroundColorAttributeName,
    NSParagraphStyleAttributeName,
)
import sys

if len(sys.argv) < 4:
    print("Usage: render_greek_texture.py <glyph> <name> <out>", file=sys.stderr)
    sys.exit(1)

glyph = sys.argv[1]
name = sys.argv[2]
out_path = sys.argv[3]

SIZE = 256

image = NSImage.alloc().initWithSize_(NSSize(SIZE, SIZE))
image.lockFocus()

bg = NSColor.colorWithCalibratedRed_green_blue_alpha_(11 / 255, 17 / 255, 34 / 255, 1.0)
bg.set()
NSBezierPath.fillRect_(NSRect(((0, 0), (SIZE, SIZE))))

font_size = SIZE * 0.32
font = NSFont.fontWithName_size_("HelveticaNeue-Bold", font_size)
if font is None:
    font = NSFont.boldSystemFontOfSize_(font_size)

paragraph = NSMutableParagraphStyle.alloc().init()
paragraph.setAlignment_(1)  # center

attrs = {
    NSFontAttributeName: font,
    NSForegroundColorAttributeName: NSColor.colorWithCalibratedRed_green_blue_alpha_(250 / 255, 204 / 255, 21 / 255, 1.0),
    NSParagraphStyleAttributeName: paragraph,
}
attributed = NSAttributedString.alloc().initWithString_attributes_(glyph, attrs)
text_size = attributed.size()
rect = NSRect(((0, (SIZE - text_size.height) / 2 - SIZE * 0.02), (SIZE, text_size.height)))
attributed.drawInRect_(rect)

image.unlockFocus()

tiff = image.TIFFRepresentation()
bitmap = NSBitmapImageRep.imageRepWithData_(tiff)
png = bitmap.representationUsingType_properties_(NSPNGFileType, None)
if not png:
    print(f"Failed to render texture for {name}", file=sys.stderr)
    sys.exit(2)

png.writeToFile_atomically_(out_path, True)
print(f"Created {out_path}")
